lib/util/include/hse_util/compiler.h

```diff
@@ -115,6 +115,13 @@
 #define HSE_WARN_UNUSED_RESULT
 #endif
 
+#if defined(__has_include) && __has_include(<stdnoreturn.h>)
```

Comment:
We have been doing these types of checks in Meson, so it should probably live there for consistency. lib/meson.build, but perhaps we should move the checks out of Meson and inline them?

Reply:
I followed the recipe for `__has_include(<threads.h>)` which is in this same file.

Reply:
I guess using `cc.compiles()` is more robust because that essentially proves it compiles versus this which simply proves the include file is available.